### PR TITLE
Docs: Fix approx. 600 typos

### PR DESCRIPTION
**Pull request by stakx**
This PR aims to fix many (if not most) of the typos that have accumulated in ImageResizer. Take special note of the following:

* Abbreviations such as PNG, JPEG, HTTP, URL are capitalized consistently, but only where they do not refer to (query-) string literals or enum values.
* Most corrections are strictly in comments and thus can't have side effects. Some, however, are in code (more specifically, in exception or log messages).
* Some corrections are made in (I guess?) contributed code or code from a third party (e.g. in the FreeImage wrapper). Perhaps the corrections should be made in the original code repository, and not here?

### CODE REVIEW

**Comment by stakx** [comment]
Not quite sure what goes wrong with the AppVeyor build, since this is essentially a no-code commit. If I can do anything to help resolve this, please let me know.

**Comment by lilith** [comment]
Perhaps a using statement got clobbered? Trying to compile AppVeyor.sln in VS2015 usually catches 95% of errors. 

I'm currently running MemTest86 on my windows box. It'll be daylight before I can open this branch in an IDE.   I can't wait for Project Rider! An xplat .NET IDE that *just might* be able to handle my solution files for ImageResizer.

**Comment by stakx** [comment]
I was under the impression that some NuGet packages don't get restored properly (types are not found). IIRC the build errors are slightly different each time; I amended the commit a time or two, so the build ran more than once.

On my way to work now where we still have VS2015, so I'll try and see what happens when I run a build using that.

It's indeed a pity that cross-platform or just large builds are currently so difficult to do with VS.

**Comment by stakx** [comment]
@nathanaeljones, here's what I found so far regarding the failed AppVeyor build. Let me know if there's anything else I can do to help.

### TL;DR: ###
I still suspect **(a)** that the AppVeyor build fails because of an incomplete or failed NuGet package restore, and **(b)** that the `Tools\nuget.exe` included in your repo (which is only a suspicious 24 KB in size) might be to blame for this. (And **(c)** I don't see anything in my commit that could even remotely influence this part of the build process.)

You can perhaps stop reading at this point and check whether something might indeed be wrong with that `Tools\nuget.exe` on AppVeyor. I don't claim to understand the whole build process in detail. Below is just a brief summary of how I arrived at the above suspicions.

### (a): Incomplete or failed NuGet package restore as the likely cause for the failing build: ###

The [AppVeyor build log](https://ci.appveyor.com/project/imazen/resizer/build/4.1.3.1017/job/s7qj1319d2q2ioie) contains lots of error messages of these kinds:

* _"This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them."_
* _"Could not resolve this reference. Could not locate the assembly "[…]". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors."_
* _"The type or namespace name '[…]' could not be found (are you missing a using directive or an assembly reference?)"_

### (b): Why `Tools\nuget.exe` might be the cause of this: ###

I've tried to execute the same steps on my machine as AppVeyor. I've cloned the repo using the same commands, I've run `Tools\builder prepare`... and got these first two error messages:

```none
Unable to find version '3.26.1' of package 'FAKE'.
Unable to find version '2.0.0' of package 'xunit.runners'.
```

These messages are caused by the command `nuget restore ..\AppVeyor.sln` inside `Tools\builder.bat` (on line 26). I discovered that your repo contains its own `Tools\nuget.exe` (only 24 KB!?) which gets used here and uses `.nuget\packages.config`.

When I ran a `msbuild AppVeyor.sln` at this point, I would get the same errors messages as the AppVeyor build log shows.

Next, I performed the same package restore with my system-wide `nuget.exe` (instead of `Tools\nuget.exe`) to perform the restore, it worked as it should:

```none
Restoring NuGet package xunit.runners.2.0.0.
Restoring NuGet package FAKE.3.26.1.
[…]
Installed:
    44 package(s) to packages.config projects
```

After this, `msbuild AppVeyor.sln` succeeds without any errors.

### (c): Is there something in this PR that causes the problem? ###

I don't think so. I only changed several "readme" `.md` and `.txt` files, as well as `.cs` source code files. (There's one exception: I've changed one line in a `.csproj` file due to a file rename). I fail to see how changes in such files could affect the NuGet package restore, which happens before compilation (thus ruling out modifications in `.cs` files) and is not related to any text files.

**Comment by lilith** [comment]
I think you're right on all counts.  I think all this PR did is invalidate some cache somewhere.  I did review the changes and they look good.

I have replacement ddr4 and my windows box is back up as of last night, so I'll try to get this CI issue troubleshooted and merged soon.

**Comment by lilith** [comment]
Rebased and merged as 2be2ad1
